src/mgmt/srv.c

```diff
@@ -61,86 +61,102 @@ static struct daos_rpc_handler mgmt_handlers[] = {
 #undef X
 
 static void
-process_daos_request(Proto__DaosRank *rank, Proto__DaosResponse *response)
+process_killrank_request(Drpc__Call *drpc_req, Proto__DaosResponse *daos_resp)
 {
+	Proto__DaosRank	*pb_rank = NULL;
+
+	proto__daos_response__init(daos_resp);
+
+	/* Unpack the daos request from the drpc call body */
+	pb_rank = proto__daos_rank__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (pb_rank == NULL) {
+		daos_resp->status = PROTO__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
```

Comment:
Just a suggestion: Maybe the status should be something more specific like invalid rank? Though perhaps it would be confused with the case where the rank number inside the struct does not refer to an actual rank in the system (which I assume we'd find out when we actually act on the request).

Reply:
my consideration was what you described, this is a problem with decoding the payload which is technically unknown,as opposed to deciding that the rank passed in the request doesn't exist on the relevant pool

Reply:
we could obviously add more error codes if we wanted

Reply:
My only hesitation with "unknown" is we could convey something about what went wrong in this case - the payload is incompatible, or got corrupted somehow while being sent over the wire. I don't think it's a big deal at this point, though.